.cascade-code/Chapman.edu/test-section/tatwell/fast-omni-nav/omni_nav.vtl

```diff
@@ -0,0 +1,513 @@
+        <span>Social</span>
+      </a>
+    </li>
+  </ul>
+#end
+
+#macro ( buildOffCanvasNavSecondaryMenu )
+  ## This is all based on existing code found in old Omni-Nav format.
+  #set ( $umbrellaPath = $pageUmbrellaCategory[0] )
+  #set ( $parentSiblingsSelector = "//system-index-block/system-folder[path = '/${umbrellaPath}']/system-folder | //system-index-block/system-folder[path = '/${umbrellaPath}']/system-page[name != 'index']" )
+  #set ( $parentSiblings = $_XPathTool.selectNodes($indexBlock, $parentSiblingsSelector) )
+
+  <ul class="level-1">
+    #if ( $parentSiblings.size() > 0 )
+      #foreach ( $sibling in $parentSiblings )
+        #buildSecondaryMenuItem( $sibling )
+      #end
+    #else
+      #set ( $pageSiblingsPath = "//system-folder[@current = 'true']/system-folder | //system-folder[@current = 'true']/system-page[name != 'index']" )
+      #set ( $pageSiblings = $_XPathTool.selectNodes($indexBlock, $pageSiblingsPath) )
+      #foreach ( $sibling in $pageSiblings )
+        #buildSecondaryMenuItem( $sibling )
+      #end
+    #end
+  </ul>
+#end
+
+#macro ( buildSecondaryMenuItem $siblingNode )
+  ## This is based on showPage macro from previous velocity format.
+  ## It builds a menu link for the secondary menu. It will either be a link or a
+  ## drop-down menu with multiple links.
+  #set ( $hideSettingSelector = "dynamic-metadata[1][value='Yes'] or system-page[name = 'index']/dynamic-metadata[1][value='Yes']" )
+  #set ( $metaHideSetting = $_XPathTool.selectNodes($siblingNode, $hideSettingSelector) )
+  #set ( $isVisible = $metaHideSetting == [false] )
+
+  ## If hidden, continue. But velocity doesn't have a continue operator:
+  ## https://stackoverflow.com/questions/7530019/apache-velocity-foreach-loop-continue
+  #if ( $isVisible )
+    #set ( $nodeType = $siblingNode.getName() )
+    #set ( $link = $siblingNode.getChild("link").value )
+
+    #if ( $nodeType == 'system-page' )
+      #set ( $displayName = $_EscapeTool.xml($siblingNode.getChild("display-name").value) )
+      <li><a href="$link">$displayName</a></li>
+
+    #elseif ( $nodeType == 'system-folder' )
+      #set ( $folderIndexPage = $_XPathTool.selectSingleNode($siblingNode, "system-page[name = 'index']") )
+      #set ( $displayName = $_EscapeTool.xml($folderIndexPage.getChild("display-name").value) )
+      #set ( $children = $_XPathTool.selectNodes($siblingNode, "system-page[name != 'index'] | system-folder") )
+
+      ## Count hidden children to see if we need to show toggle button.
+      #set ( $hiddenChildCounter = 0 )
+      #set ( $hiddenChildSelector = "dynamic-metadata[1][value='Yes'] | system-folder/dynamic-metadata/value" )
+      #foreach ( $child in $children )
+        #if ( $_XPathTool.selectSingleNode($child, $hiddenChildSelector) )
+          #set ( $hiddenChildCounter = $hiddenChildCounter + 1 )
+        #end
+      #end
+      #set ( $hasVisibleChildren = $children.size() > 0 && $hiddenChildCounter < $children.size() )
+
```

Comment:
This is what we use in the left nav format for hidden children:
```
#set ( $hiddenChildren = $_XPathTool.selectNodes($sibling, "system-page[name != 'index']/dynamic-metadata[1][value='Yes'] | system-folder[dynamic-metadata[1][value='Yes']]") )
        
#if ( $children.size() > $hiddenChildren.size() )
  #set ( $siblingHasChildren = true )
#else
  #set ( $siblingHasChildren = false )
#end
```
Would this not work here as well?

Also I do like `#set ( $hasVisibleChildren = $children.size() > 0 && $hiddenChildCounter < $children.size() )` and something like that would replace the if/else in the code I pasted above but there was a comment (from Mandy maybe?) that said it doesn't work to set a variable equal to a conditional. This is a comment from the old LeftNavNoSiblings format: `### DIDN't WORK: set ( $siblingHasChildren = $children.size() > 0 )`

Reply:
It did cross my mind that there might be issues with that conditional expression. It seemed to work in my tests. But I'll explicitly test and report back.

`selectNodes` code does look cleaner than the loop. I'll change this.

Reply:
remember that we hide all of the following: children/sibling pages marked as Hidden, and the index pages in the subfolders if marked as Hidden (remember they are children of the index page in parent folder), AND if a folder itself is marked as Hidden then no pages or subfolders/index-pages under it should show.

Reply:
@kaylaziegler It looks like conditional assignment does work. See latest changeset:

- https://github.com/chapmanu/cascade-assets/pull/243/commits/71a63c8d250e1778ca103039beacfe2d319a8eca#diff-c9f496e65abd2235fce97c7989dae13bR69

Results can be viewed here (bottom of page):

- https://dev-www.chapman.edu/test-section/tatwell/fast-omni-nav/demo.html.aspx